public/app/plugins/datasource/influxdb/components/QueryEditorModeSwitcher.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { useEffect, useState } from 'react';
+import { Button, ConfirmModal } from '@grafana/ui';
+
+type Props = {
+  isRaw: boolean;
+  onChange: (newIsRaw: boolean) => void;
+};
+
+export const QueryEditorModeSwitcher = ({ isRaw, onChange }: Props): JSX.Element => {
+  const [isModalOpen, setModalOpen] = useState(false);
+
+  useEffect(() => {
+    // if the isRaw changes, we hide the modal
+    setModalOpen(false);
+  }, [isRaw]);
+
+  if (isRaw) {
+    return (
+      <>
+        <Button
```

Comment:
Could we use [ConfirmButton](https://developers.grafana.com/ui/latest/index.html?path=/docs/buttons-confirmbutton--basic) instead?